File: apps/mobile-android/app/src/main/kotlin/ai/tide/app/data/models/Message.kt
package ai.tide.app.data.models

import java.util.UUID
import java.util.Date

// MARK: - Message Model
data class Message(
    val id: String = UUID.randomUUID().toString(),
    val content: String,
    val role: MessageRole,
    val timestamp: Date = Date(),
    val status: MessageStatus = MessageStatus.SENT,
    val actionPreview: ActionPreview? = null,
    val suggestions: List<String>? = null,
    val aiSummary: String? = null
)

// MARK: - Message Role
enum class MessageRole {
    USER,
    ASSISTANT,
    SYSTEM
}

// MARK: - Message Status
enum class MessageStatus {
    SENDING,
    SENT,
    DELIVERED,
    READ,
    FAILED
}

// MARK: - Action Preview
data class ActionPreview(
    val title: String,
    val description: String,
    val actionType: ActionType,
    val requiresConfirmation: Boolean,
    val isConfirmed: Boolean = false
) {
    enum class ActionType {
        SCHEDULE_EVENT,
        SEND_EMAIL,
        CREATE_TASK,
        UPDATE_CALENDAR,
        DELEGATE_TASK,
        ANALYZE_DOCUMENT
    }
}

File: apps/app/app/Models/Message.swift
import Foundation

// MARK: - Message Model
struct Message: Identifiable, Codable, Hashable {
    let id: String
    var content: String
    let role: MessageRole
    let timestamp: Date
    var status: MessageStatus
    var actionPreview: ActionPreview?
    var suggestions: [String]?
    var aiSummary: String?

    init(
        id: String = UUID().uuidString,
        content: String,
        role: MessageRole,
        timestamp: Date = Date(),
        status: MessageStatus = .sent,
        actionPreview: ActionPreview? = nil,
        suggestions: [String]? = nil,
        aiSummary: String? = nil
    ) {
        self.id = id
        self.content = content
        self.role = role
        self.timestamp = timestamp
        self.status = status
        self.actionPreview = actionPreview
        self.suggestions = suggestions
        self.aiSummary = aiSummary
    }
}

// MARK: - Message Role
enum MessageRole: String, Codable {
    case user
    case assistant
    case system
}

// MARK: - Message Status
enum MessageStatus: String, Codable {
    case sending
    case sent
    case delivered
    case read
    case failed
}

// MARK: - Action Preview
struct ActionPreview: Codable, Hashable {
    let title: String
    let description: String
    let actionType: ActionType
    let requiresConfirmation: Bool
    var isConfirmed: Bool

    enum ActionType: String, Codable {
        case scheduleEvent
        case sendEmail
        case createTask
        case updateCalendar
        case delegateTask
        case analyzeDocument
    }
}

// MARK: - Message Extensions
extension Message {
    var timeString: String {
        let formatter = DateFormatter()
        formatter.timeStyle = .short
        return formatter.string(from: timestamp)
    }

    var dateString: String {
        let formatter = DateFormatter()
        formatter.dateStyle = .medium
        return formatter.string(from: timestamp)
    }
}
